src/generator.cs

```diff
@@ -3650,7 +3654,11 @@ void GetReturnsWrappers (MethodInfo mi, MemberInformation minfo, Type declaringT
 		} else if (mai.Type.IsArray){
 			Type etype = mai.Type.GetElementType ();
 			if (minfo != null && minfo.is_bindAs) {
-				var bindAsT = GetBindAsAttribute (minfo.mi).Type.GetElementType ();
+				var bindAttrType = GetBindAsAttribute (minfo.mi).Type;
+				if (!bindAttrType.IsArray) {
+					throw new BindingException (1071, true, "The BindAs type for the member \"{0}.{1}\" must be an array when the member's type is an array.", mai.Type.Name, minfo.mi.Name);
```

Comment:
Same, the type name is the member's type, I think this is right:

```suggestion
					throw new BindingException (1071, true, "The BindAs type for the member \"{0}.{1}\" must be an array when the member's type is an array.", minfo.mi.DeclaringType.FullName, minfo.mi.Name);
```